### PR TITLE
Add Plan Trip Button to Advanced Mode Settings Page

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**
- Adds 'back' and "plan trip" buttons to the advanced trip form
- Moves around some of the planQuery logic to allow it to be accessed by the advanced trip form
- Adds a toast notification when the advanced settings close, indicating to the user that the settings have been saved

image

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/components/form/advanced-settings-panel.tsx [hunk 1]
```diff
@@ -28,20 +20,30 @@ import PageTitle from '../util/page-title'
...
 } from './util'
+import { AppReduxState } from '../../util/state-types'
 import { setModeButtonEnabled } from './batch-settings'
 
+import { blue, getBaseColor } from '../util/colors'
+import { ComponentContext } from '../../util/contexts'
+
 import styled from 'styled-components'
 
+const baseColor = getBaseColor()
```
Comment (josh-willis-arcadis) [inline lib/components/form/advanced-settings-panel.tsx:43]: not working for me. potential timing issues with when the function is called and css is loaded?

File: lib/components/form/util.tsx [hunk 2]
```diff
@@ -69,3 +72,45 @@ export const setModeButton =
       encodeQueryParams(modesQueryParamConfig, { modeButtons: newButtons })
     )
   }
+
+export const alertUserTripPlan = (
+  intl,
+  currentQuery,
+  onPlanTripClick,
+  routingQuery
```
Comment (josh-willis-arcadis) [inline lib/components/form/util.tsx:80]: getting eslint warnings and errors for no return type on function and implicit 'any' types.

File: lib/components/mobile/batch-search-screen.tsx [hunk 3]
```diff
@@ -21,6 +21,8 @@ import SwitchButton from '../form/switch-button'
 import MobileContainer from './container'
 import MobileNavigationBar from './navigation-bar'
 
+import toast from 'react-hot-toast'
```
Comment (josh-willis-arcadis) [inline lib/components/mobile/batch-search-screen.tsx:24]: sort imports

**Review by josh-willis-arcadis — CHANGES_REQUESTED** [review]
Looking good. Not sure what is going on with getBaseColor yet.

**Review by daniel-heppner-ibigroup — CHANGES_REQUESTED** [review]
Buttons look good, functionality good, just improve the types please!

**Review by daniel-heppner-ibigroup — APPROVED** [review]
I fixed both my and Josh's comments, @josh-willis-arcadis please look closely at my comments too to make sure they're actually fixed, since I was the one who fixed them I'd like to make sure they get a second look.

File: lib/components/form/advanced-settings-panel.tsx (hunk 1 above)
Comment (daniel-heppner-ibigroup) [reply lib/components/form/advanced-settings-panel.tsx]: You were right, the CSS variable isn't loaded yet when this got called. I moved it around and used the vanilla CSS variable in the styled components, no need to take a CSS variable, make it a JS variable, just to put it back in the CSS!

File: lib/components/form/advanced-settings-panel.tsx [hunk 4]
```diff
@@ -79,26 +73,57 @@ const HeaderContainer = styled.div`
...
-const accentColor = baseColor || blue[900]
+
+const PlanTripButton = styled.button`
+  align-items: center;
+  display: flex;
+  justify-content: center;
+  gap: 0.5em;
+  background-color: var(--main-base-color, ${blue[900]});
+  border: 0;
+  width: 45%;
```
Comment (josh-willis-arcadis) [inline lib/components/form/advanced-settings-panel.tsx:84]: super nit picky but from widths 992px - 1058px the trip text gets pushed to a new line. change the width to like 47% to fit the text on a single line.

**Review by josh-willis-arcadis — APPROVED** [review]
The changes look great! One thing I noticed is that if you uncheck every mode and then click "Plan Trip," you get an error message, followed by the "Preferences Saved" toast, and the advanced panel closes. If there is an error, I believe we shouldn't close the panel or display the toast.

File: lib/components/form/advanced-settings-panel.tsx (hunk 4 above)
Comment (amy-corson-ibigroup) [reply lib/components/form/advanced-settings-panel.tsx:84]: Really clean, thank you!!

**Comment by amy-corson-ibigroup** [comment]
@josh-willis-arcadis  really good point about the toast. I went ahead and adjusted when the toast displays - now it will only happen when the user closes the panel and returns to the home page, and not when they click "plan trip"

**Comment by amy-corson-ibigroup** [comment]
@josh-willis-arcadis @daniel-heppner-ibigroup thank you both so much for the help on this one!

**Pull request merged**